presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -926,6 +928,16 @@ private InputStream openStream(Path path, long start)
                                             // ignore request for start past end of object
                                             return new ByteArrayInputStream(new byte[0]);
                                         case SC_FORBIDDEN:
+                                            // S3 object can transition to Glacier storage class due to lifecycle policy
+                                            // It remains in s3 bucket and causes get object request to fail and throws exception as
+                                            // "com.amazonaws.services.s3.model.AmazonS3Exception: The operation is not valid for the object's storage class (Service: Amazon S3; Status Code: 403 ..."
+                                            // Verify exception is caused by trying to get Glacier object and skip this object
+                                            ListObjectsV2Result result = s3.listObjectsV2(host, keyFromPath(path));
+                                            boolean isGlacierObject = result.getObjectSummaries().stream()
```

Comment:
Why don't we do this filtering in `listPrefix()`? We already have that information available there.

Reply:
Will move the filtering to listPrefix( ).